### PR TITLE
github/ci: New CI framework

### PR DESCRIPTION
**Pull request by phlax**
README: https://github.com/envoyproxy/envoy/blob/48afe3a868c3a175c393d5feca285cf1345a6c7a/.github/workflows/README.md

config: https://github.com/envoyproxy/envoy/blob/48afe3a868c3a175c393d5feca285cf1345a6c7a/.github/config.yml

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: ci/do_ci.sh [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 # Run a CI build/test target, e.g. docs, asan.
 
+# ARB CHANGE
```
Comment (phlax) [inline ci/do_ci.sh:5]: revert this file

**Review by wbpcode — DISMISSED** [review]
Blinded LGTM. Although I had a talk with @phlax offline, it's still almost impossible for me to give a reasonable review to this super large PR.

The annoying point it that we can only verify the CI after we merge it. Anyway, let's do it.

**Comment by phlax** [comment]
@wbpcode ive had a go at simplifying the filters in _finish.yml and the upstream ci/request action - hopefully they are a bit cleaner/clearer

File: .github/workflows/_precheck_deps.yml [hunk 2]
```diff
@@ -1,46 +1,50 @@
...
+    name: ${{ matrix.target }}
     with:
-      target: ${{ matrix.target }}
-      rbe: ${{ matrix.rbe }}
       bazel-extra: '--config=rbe-envoy-engflow'
-      cache-build-image: ${{ inputs.build-image-ubuntu }}
-      repo-ref: ${{ inputs.repo-ref }}
-      catch-errors: true
+      cache-build-image: ${{ fromJSON(inputs.request).request.build-image.default }}
+      # catch-errors: true
```
Comment (phlax) [inline .github/workflows/_precheck_deps.yml:34]: probs best to just remove

**Comment by phlax** [comment]
> it's hard to review it thoroughly and understand how it works compared to previous CI

i added a README to try and cover the headline bits - but we can defo document this better as we go

**Review by abeyad — DISMISSED** [review]

**Pull request merged**